### PR TITLE
Guest search hotel

### PR DESCRIPTION
**Pull request by Arribas12**
Being a guest user you can search for the available hotels from a selected date. You can also logout.
There are some javadocs on IHotelDao.
HotelDaoTest missing

### CODE REVIEW

**Review by iapellaniz — CHANGES_REQUESTED** [review]
- Functionality looks good, although the UI is not that intuitive to use (we can sort it out in another PR)
- Merge master into the branch to see if it works and implement new changes
- Missing tests
- Lots of uncommented code

File: src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		this.screenHeight = screenHeight;
+		
+		viewHotel = new JButton("View hotels");
+		viewHotel.setSize(100, 30);
+		viewHotel.addActionListener(new ActionListener() {
+			
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				if(calendar.getSelectedDate() == null) {
+					JOptionPane.showMessageDialog(null, "Please select a date", "Error", JOptionPane.ERROR_MESSAGE);
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java]: Use `LocaleManager` when displaying text in the client UI

File: src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+				}
+				else {
+					controller.setCurrentHotels();
+					ArrayList<HotelDTO> retrievedHotels = controller.retrieveHotels(calendar.getSelectedDate().toString());
+					if(retrievedHotels == null || retrievedHotels.size() == 0) {
+						JOptionPane.showMessageDialog(null, "There are no hotels available", "Error", JOptionPane.ERROR_MESSAGE);
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java]: Use `LocaleManager` when displaying text in the client UI

File: src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+									+ "-" + seasonEndingDate});
+
+							controller.setCurrentHotels(hotel);
+						}
+					}
+				}
+			}
+		});
+		
+		confirm = new JButton("Confirm");
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java]: Use `LocaleManager` when displaying text in the client UI

File: src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+		logout = new JButton("Log out");
+		logout.setSize(100, 30);
+		logout.setBackground(Color.white);
+		logout.addActionListener(new ActionListener() {
+			
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				try {
+					controller.logOut();
+					JOptionPane.showMessageDialog(null,  "Successfully logged out", "Logged out",JOptionPane.INFORMATION_MESSAGE);
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/client/gui/HotelGuestSearchingPanel.java]: Use `LocaleManager` when displaying text in the client UI

File: src/main/java/es/deusto/spq/server/data/jdo/Hotel.java [hunk 5]
```diff
@@ -15,6 +16,7 @@
 	private String location;
 	private Timestamp seasonStart;
 	private Timestamp seasonEnding;
+	private ArrayList<Room> room;
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/server/data/jdo/Hotel.java]: Try to use `List` instead of `ArrayList`

**Review by iapellaniz — COMMENTED** [review]
- Please use `LocaleManager` for displaying UI messages.

- I see you're using your own UI system instead of views. This will be problematic.
 The problem is that if you use only one `View` and inside that view you show different "Screens" with different purposes and permission levels, the system won't know how to handle a permission level change (log in, log out, ...). If you decide to keep the current system (instead of using different views for each window), you'll have to properly implement the `getViewPermission` and `getViewType` so that the correct `ViewType` and `ViewPermission` gets returned.
    - Also, please keep in mind that if the `ViewManager` needs to open another window your UI system must be able to be instantiated multiple times with different screens active.

**Review by iker-barriocanal — CHANGES_REQUESTED** [review]
There's a lot of documentation missing.

File: src/main/java/es/deusto/spq/server/data/dao/HotelDAO.java [hunk 6]
```diff
@@ -212,9 +210,9 @@ public void cleanDB() {
 				tx.rollback();
 			}
 
-			if (pm != null && !pm.isClosed()) {
-				pm.close();
-			}
+//			if (pm != null && !pm.isClosed()) {
```
Comment (iapellaniz) [inline src/main/java/es/deusto/spq/server/data/dao/HotelDAO.java]: Please, if code isn't used, remove it instead of commenting it out

**Comment by Egoes98** [comment]
First you shold merge master into this branch to resolve all the conflict and see if it works together.

-Test Fails
image
-It work fine but some changes should be made. When you signIn as a guest it should show only the Guest GUI if you log in as an admin only the admin one.
-More documentation(Look at the checklist to see what documentation you are missing):

File: src/main/java/es/deusto/spq/server/data/jdo/Room.java [hunk 7]
```diff
@@ -2,31 +2,34 @@
...
 public class Room {
 	
+	@PrimaryKey
 	private String roomId;
 	private float size; 
 	private float price; 
 	private RoomType type;
 	private boolean isOccupied;
-	private List<String> features;
+//	private List<String> features;
```
Comment (Egoes98) [inline src/main/java/es/deusto/spq/server/data/jdo/Room.java]: Remove unused code

**Review by Egoes98 — DISMISSED** [review]
-Test fails
image
-When you log in as guest you shold only see guest window and when admin only the admin. seen the execution i suposse is not like that.
-More documentation(look at the ckecklist report to see the requested javadock)

File: src/main/java/es/deusto/spq/server/data/jdo/Room.java (hunk 7 above)
Comment (Arribas12) [reply src/main/java/es/deusto/spq/server/data/jdo/Room.java]: check now if it is corrected please

**Review by iapellaniz — DISMISSED** [review]
Looking good!

Just a couple of simple changes:

- Please, remove the "Log out" buttons and dialogs, as that will now be handled by the new User system.
- When I select a row from the results table, it turns red. Is it intentional?
- Please use the Locale system! If you have doubts about it, look at #26 to see an example implementation

Thanks! If you need help about the last one just tell me

**Review by Egoes98 — APPROVED** [review]
I think its good.

**Review by iker-barriocanal — APPROVED** [review]
Try to improve your code (e.g. using `List` instead of `ArrayList` in the method's definition); and the more documentation you do the better. All in all, good.

**Pull request merged**